pkg/vpcmodel/explainabilityConnectivity.go

```diff
@@ -476,3 +482,59 @@ func (v *VPCConnectivity) getConnection(c *VPCConfig, src, dst Node) (conn *deta
 	}
 	return conn, nil
 }
+
+func (details *rulesAndConnDetails) updateRespondRules(c *VPCConfig, connQuery *connection.Set) error {
+	for _, srcDstDetails := range *details {
+		// respond rules are relevant if connection has a TCP component and non-stateful filter (NACL at the moment)
+		// are relevant for <src, dst>
+		if !respondRulesRelevant(srcDstDetails.conn, srcDstDetails.filtersRelevant) {
+			continue
+		}
+		connForResp := newTCPSet()
+		if connQuery != nil {
+			connForResp = connForResp.Intersect(connQuery)
+		}
+		respondRules, err := c.getRespondRules(srcDstDetails.src, srcDstDetails.dst, connForResp)
+		if err != nil {
+			return err
+		}
+		srcDstDetails.respondRules = respondRules
+	}
+	return nil
+}
+
+func respondRulesRelevant(conn *detailedConn, filtersRelevant map[string]bool) bool {
+	return !conn.tcpRspEnable.Union(conn.tcpRspDisable).IsEmpty() && filtersRelevant[NaclLayer]
+}
+
+// gets the NACL rules that enables/disables respond for connection conn, assuming nacl is applied
+func (c *VPCConfig) getRespondRules(src, dst Node,
+	conn *connection.Set) (respondRules *rulesConnection, err error) {
+	ingressAllowPerLayer, egressAllowPerLayer := rulesInLayers{}, rulesInLayers{}
+	ingressDenyPerLayer, egressDenyPerLayer := rulesInLayers{}, rulesInLayers{}
+	// todo: switch dst src ports of conn - to that end needs to merge the PR on connections that exports the func
+	connSwitch := conn
+	mergedIngressRules, mergedEgressRules := rulesInLayers{}, rulesInLayers{}
+	// respond: from dst to src. Thus, ingress rules: relevant only if *src* is internal, egress is *dst* is internal
+	if src.IsInternal() {
+		// respond: dst and src switched
+		ingressAllowRules, ingressDenyRules, err1 := c.getFiltersRulesBetweenNodesPerDirectionAndLayer(dst, src, connSwitch, true, NaclLayer)
+		if err1 != nil {
+			return nil, err1
+		}
+		ingressAllowPerLayer.updateRulesPerLayerIfNonEmpty(NaclLayer, ingressAllowRules)
+		ingressDenyPerLayer.updateRulesPerLayerIfNonEmpty(NaclLayer, ingressDenyRules)
+		mergedIngressRules = mergeAllowDeny(ingressAllowPerLayer, ingressDenyPerLayer)
+	}
```

Comment:
put these 5 code lines in a function, to avoid code duplication

Reply:
done